src/Monomer/Widgets/Animation/Fade.hs

```diff
@@ -127,81 +106,25 @@ animFadeOut managed = animFadeOut_ def managed
+animFadeOut_ configs managed = makeNode configs managed False
+  & L.info . L.widgetType .~ "animFadeOut"
 
 makeNode
-  :: WidgetEvent e => WidgetType -> Widget s e -> WidgetNode s e -> WidgetNode s e
-makeNode wType widget managedWidget = defaultWidgetNode wType widget
-  & L.info . L.focusable .~ False
-  & L.children .~ Seq.singleton managedWidget
-
-makeFade :: WidgetEvent e => Bool -> FadeCfg e -> FadeState -> Widget s e
-makeFade isFadeIn config state = widget where
-  widget = createContainer state def {
-    containerInit = init,
-    containerMerge = merge,
-    containerHandleMessage = handleMessage,
-    containerRender = render,
-    containerRenderAfter = renderPost
-  }
-
-  FadeState running start = state
-  autoStart = fromMaybe False (_fdcAutoStart config)
-  duration = fromMaybe 500 (_fdcDuration config)
-  period = 20
-  steps = fromIntegral $ duration `div` period
-
-  finishedReq node ts = delayedMessage node (AnimationFinished ts) duration
-  renderReq wenv node = req where
-    widgetId = node ^. L.info . L.widgetId
-    req = RenderEvery widgetId period (Just steps)
-
-  init wenv node = result where
-    ts = wenv ^. L.timestamp
-    newNode = node
-      & L.widget .~ makeFade isFadeIn config (FadeState True ts)
-    result
-      | autoStart = resultReqs newNode [finishedReq node ts, renderReq wenv node]
-      | otherwise = resultNode node
-
-  merge wenv node oldNode oldState = resultNode newNode where
-    newNode = node
-      & L.widget .~ makeFade isFadeIn config oldState
-
-  handleMessage wenv node target message = result where
-    result = cast message >>= Just . handleAnimateMsg wenv node
-
-  handleAnimateMsg wenv node msg = result where
-    widgetId = node ^. L.info . L.widgetId
-    ts = wenv ^. L.timestamp
-    startState = FadeState True ts
-    startReqs = [finishedReq node ts, renderReq wenv node]
-
-    newNode newState = node
-      & L.widget .~ makeFade isFadeIn config newState
-    result = case msg of
-      AnimationStart -> resultReqs (newNode startState) startReqs
-      AnimationStop -> resultReqs (newNode def) [RenderStop widgetId]
-      AnimationFinished ts'
-        | isRelevant -> resultEvts node (_fdcOnFinished config)
-        | otherwise -> resultNode node
-        where isRelevant = _fdsRunning state && ts' == _fdsStartTs state
-
-  render wenv node renderer = do
-    saveContext renderer
-    when running $
-      setGlobalAlpha renderer alpha
-    where
-      ts = wenv ^. L.timestamp
-      currStep = clampAlpha $ fromIntegral (ts - start) / fromIntegral duration
-      alpha
-        | isFadeIn = currStep
-        | otherwise = 1 - currStep
-
-  renderPost wenv node renderer = do
-    restoreContext renderer
+  :: WidgetEvent e
+  => [FadeCfg s e]
+  -> WidgetNode s e
+  -> Bool
+  -> WidgetNode s e
+makeNode configs managed isFadeIn = node where
+  node = animTransform_ [_fdcTransformCfg] f managed
+  f t _ = [animGlobalAlpha $ alpha t]
+  alpha t = if isFadeIn
```

Comment:
@Deltaspace0 no need to change this, but sometimes the `guard` format is more concise:

```haskell
alpha t
  | isFadeIn = currStep t
  | otherwise = 1 - currStep t
```

Reply:
In this case I see both `if` and `guard` formats readable, but personally I prefer to use guard only when there are 3 or more conditions.